src/components/editor/Document.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-underscore-dangle */
+import Vue from 'vue';
+
+export default class Document {
```

Comment:
Document is such an overloaded term. Can we consider something more specific (and easily searchable)? This looks DocumentUpdater?